README.md

```diff
@@ -1,178 +1,114 @@
-    ],
-
-    files: [
-        ./**/*.lu
-        ./**/*.lg
-        ./**/*.dialog
-    ]
-    
-    entry: "./dialogs/start.dialog"
+  services: [
+    {
+      type: "luis",
+      endpoint: ${luis_endpoint},
+      app_id: ${luis_app_id},
+      token: ${luis_token} 
+    },
+    {
+      type: "qna",
+      endpoint: ${qna_endpoint},
+      app_id: ${qna_app_id},
+      token: ${qna_token} 
+    }
+  ],
 
-}
+  files: [
+    "./**/*.lu",
+    "./**/*.lg",
+    "./**/*.dialog"
+  ]
 
+  dialogEntry: "./dialogs/start.dialog",
+  start: "node index.js"
+}
 ```
 
-.bot is basicly a project file, include all the references to all assets, and other necessary infomation to run a bot. 
-
-## Bot launcher configuration
+the .bot file is used for the following:
+- asset discovery via glob patterns
+- service discovery for luis and qna, etc
+- service credentials
+- entry points for the dialog tree and the bot's runtime.
 
-Since all assets are defined in bots. The bot launcher's config is simple, only need to know about where to find a bot.
-
-```
-/BotLauncher/CSharp/appsettings.json
-{
-    // this is the bot to be launched
-    bot: {
-        provider: "localDisk",
-        path: "../../Bots/SampleBot1/bot1.bot
-    }
-
-    // it may include some othe info to run like
-    endpoint: "localhost:3979"
-}
-```
+**note:** a goal to consider is the service credentials being *templated* in this file. we've made assumptions in the past about where these credentials should be stored and their shape.
 
 ## Composer configuration
 
 ### config.json
 
-config.json in composer is where to specfic 
-1. which bot we are editing (if we follow vscode convention, this should be in .composer, but let's make it simple here)
-2. which launcher we are using
-
 ```
-/Composer/config.json 
 {
-    // this is the bot we are editing\composing
-    bot: {
-        provider: "localDisk"
-        path: "../../Bots/SampleBot1/bot1.bot"
-    }
-    
-    launcher: {
-        startCommand: "cp launcher.json '../../BotLauncher/CSharp/appsetings.json' && cd '../../BotLauncher/CSharp' && dotnet run"
+  // the path to the BotLauncher root
+  launcherDir: "../../BotLauncher",
+  bots: [
+    //an array of bots to load in this application
+    {
+      path: "../../Bots/SampleBot1/bot1.bot",
+      secrets: "../path/to/appsettings.json" // the path to credentials necessary to run the bot (and template into the .bot file)
+    },
+    {
+      path: "../../Bots/SampleBot2/bot2.bot",
+      secrets: "../path/to/local.env"
     }
+  ]
 }
 ```
 
-It's very self-explained, the "bot" section tell the composer where the bot is, the "launcher" section tell the composer how to play with the launcher. 
-
-### launcher4[CS\NODE].json
-
-there is also a `launcher4CS.json` inside composer, which is designed to override the config inside launcher, to make sure the launcher is started the way composer wants.
-
-``` 
-/Composer/launcher4CS.json (same schema as /BotLauncher/CSharp/appsettings.json)
```

Comment:
i don't like that the schema is duplicated, we should reference one schema and pass it around to who needs it

Reply:
the reason it's duplicated is that, we design the way that bot launcher doesn't need to know the exist of composer. the composer talk to launcher following all launcher's conventions, so we can prepare a config file for each launcher, and start the launcher against that file.